File: Lista IV (JAVA) - Aula 10/J2_04/DVD.java
public class DVD extends Produto {
    private int duracao;

    DVD(){}

    DVD(String nome, double preco, int duracao){
        super(nome, preco);
        this.setDuracao(duracao);
    }

    public void setDuracao(int duração) {
        this.duracao = duração;
    }

    public int getDuracao() {
        return duracao;
    }

    public String toString(){
       
        return super.toString() + this.duracao + " minutos";
    }
}

File: Lista V (C++) - Aula 22/C1_06/DVD.cpp
#include <iostream>
#include "DVD.h"

//Construtor
DVD::DVD(const string& nome, double preco, int duracao): Produto(nome, preco){
    this->duracao = duracao;
}

//Métodos
void DVD::print(ostream &out) const {
	Produto::print(out); //chama a função de nome identico pertencente à produto
	out << " - " << duracao << " minutos"; //acrescenta a exclusividade de DVD
}

ostream& operator<< (ostream &out, DVD &dvd) {
	dvd.print(out);
	return out;
}
